File: app/src/main/java/com/faramarz/tictacdev/sharedpreferences/utils/EnumGender.java
package com.faramarz.tictacdev.sharedpreferences.utils;

/**
 * Created by Reza Amozadeh on 4/9/2018.
 */

public enum EnumGender {

    UNDEFINE("UNDEFINE", -1),
    MALE("MALE", 0),
    FEMALE("FEMALE", 1),
    ;

    private String valueStr;

    private Integer value;

    EnumGender(String valueStr, Integer value) {
        this.valueStr = valueStr;
        this.value = value;
    }

    public static EnumGender get(String value) {
        if (value == null) {
            return UNDEFINE;
        }

        EnumGender[] arr$ = values();
        for (EnumGender val : arr$) {
            if (val.valueStr.equalsIgnoreCase(value.trim())) {
                return val;
            }
        }

        return UNDEFINE;
    }

    public static EnumGender get(Integer value) {

        if (value == null) {
            return UNDEFINE;
        }

        EnumGender[] arr$ = values();
        for (EnumGender val : arr$) {
            if (val.value == value) {
                return val;
            }
        }

        return UNDEFINE;
    }

    public String getValueStr() {
        return valueStr;
    }

    public Integer getValue() {
        return value;
    }

    public void setValueStr(String valueStr) {
        this.valueStr = valueStr;
    }
}

File: app/src/main/java/com/faramarz/tictacdev/sharedpreferences/enum_package/EnumGender.kt
package com.faramarz.tictacdev.sharedpreferences.enum_package

enum class EnumGender private constructor(var valueStr: String?, val value: Int?) {

    UNDEFINE("UNDEFINE", -1),
    MALE("MALE", 0),
    FEMALE("FEMALE", 1);


    companion object {

        operator fun get(value: String?): EnumGender {
            if (value == null) {
                return UNDEFINE
            }

            val `arr$` = values()
            for (`val` in `arr$`) {
                if (`val`.valueStr!!.equals(value.trim { it <= ' ' }, ignoreCase = true)) {
                    return `val`
                }
            }

            return UNDEFINE
        }

        operator fun get(value: Int?): EnumGender {

            if (value == null) {
                return UNDEFINE
            }

            val `arr$` = values()
            for (`val` in `arr$`) {
                if (`val`.value === value) {
                    return `val`
                }
            }

            return UNDEFINE
        }
    }
}
